### PR TITLE
Addition of "Two City Scheduling" 3D Dynamic Programming Question

### LINKED ISSUE #129: [Feature Request] Addition of "Two City Scheduling" 3D Dynamic Programming Question
### Describe the solution you'd like
A comprehensive explanation of the solution, which includes:

Dynamic Programming Approach:
A clear outline of the dynamic programming solution, possibly using a 3D DP array to track the costs associated with sending a specific number of people to each city.

Base Case:
Explanation of how to initialize the DP array based on the cost of sending 0 people to each city.

Transition:
Details on how to fill in the DP array, including the conditions for choosing to send a person to either city A or B, and how this affects the overall cost.

Final Calculation:
An explanation of how to derive the minimum total cost from the DP array once it has been fully populated.

Time and Space Complexity:
An analysis of the time and space complexity of the solution.

Code Implementation:
A complete code solution in a programming language (C++, Java, Python, Javascript, Go) that implements the dynamic programming approach.

Problem link:
Link to solve the problem. 

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by J-B-Mugundh**
### Description
This PR introduces a comprehensive solution to the Two City Scheduling Problem, utilizing a dynamic programming approach to optimize the assignment of people to two different cities while minimizing travel costs.

Fixes #129 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/dynamic-programming/Two-City-Scheduling-3D-DP.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
+# Two City Scheduling Problem
```
Comment (ajay-dhangar) [inline docs/dynamic-programming/Two-City-Scheduling-3D-DP.md]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by J-B-Mugundh** [comment]
@ajay-dhangar Updated now!

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**